syntax = "proto3";

package deploything.v1;

message AgentSnapshot {
  repeated ContainerStatus container_status = 1;
}

message ContainerStatus {
  optional string id = 1;
  optional string name = 2;
  optional string image_ref = 3;
  optional ContainerState container_state = 4;
}

enum ContainerState {
  CONTAINER_STATE_UNSPECIFIED = 0;
  CONTAINER_STATE_RUNNING = 1;
  CONTAINER_STATE_EXITED = 2;
}
